sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/planner/BeamQueryPlanner.java

```diff
@@ -148,12 +156,39 @@ public BeamRelNode convertToBeamRel(String sqlStatement)
               .replace(BeamLogicalConvention.INSTANCE)
               .replace(root.collation)
               .simplify();
-      beamRelNode = (BeamRelNode) planner.transform(0, desiredTraits, root.rel);
```

Comment:
planner.transform runs the volcano planner, you've just broken it out inline here. This is not safe, as JDBC does not go through this path. If you want to feed rules into the not JDBC path, you can do that through `ruleSets` above.